x/concentrated-liquidity/fees_test.go

```diff
@@ -951,3 +952,127 @@ func (s *KeeperTestSuite) TestUpdateFeeAccumulatorPosition() {
+	var (
+		five              = sdk.NewDec(5)
+		belowDefaultPrice = DefaultCurrPrice.Sub(five)
+		aboveDefaultPrice = DefaultCurrPrice.Add(five)
+		onePercentFee     = sdk.NewDecWithPrec(1, 2)
+	)
+
+	sqrtBelowDefaultPrice, err := belowDefaultPrice.ApproxSqrt()
+	suite.Require().NoError(err)
+
+	sqrtAboveDefaultPrice, err := aboveDefaultPrice.ApproxSqrt()
+	suite.Require().NoError(err)
+
+	tests := map[string]struct {
+		currentSqrtPrice         sdk.Dec
+		nextTickSqrtPrice        sdk.Dec
+		sqrtPriceLimit           sdk.Dec
+		amountIn                 sdk.Dec
+		amountSpecifiedRemaining sdk.Dec
+		swapFee                  sdk.Dec
+
+		expectedFeeCharge sdk.Dec
+		expectPanic       bool
+	}{
+		"current sqrt price == next tick sqrt price -> charge fee on amount in": {
+			currentSqrtPrice:         DefaultCurrSqrtPrice,
+			nextTickSqrtPrice:        DefaultCurrSqrtPrice,
+			sqrtPriceLimit:           sqrtAboveDefaultPrice,
+			amountIn:                 sdk.NewDec(100),
+			amountSpecifiedRemaining: five,
+			swapFee:                  onePercentFee,
+
+			expectedFeeCharge: sdk.OneDec(),
+		},
+		"current sqrt price == sqrt price limit -> charge fee on amount in": {
+			currentSqrtPrice:         DefaultCurrSqrtPrice,
+			nextTickSqrtPrice:        sqrtAboveDefaultPrice,
+			sqrtPriceLimit:           DefaultCurrSqrtPrice,
+			amountIn:                 sdk.NewDec(100),
+			amountSpecifiedRemaining: five,
+			swapFee:                  onePercentFee,
+
+			expectedFeeCharge: sdk.OneDec(),
+		},
+		"current sqrt price == next tick sqrt price && current sqrt price == sqrt price limit -> charge fee on amount in": {
+			currentSqrtPrice:         DefaultCurrSqrtPrice,
+			nextTickSqrtPrice:        DefaultCurrSqrtPrice,
+			sqrtPriceLimit:           DefaultCurrSqrtPrice,
+			amountIn:                 sdk.NewDec(100),
+			amountSpecifiedRemaining: five,
+			swapFee:                  onePercentFee,
+
+			expectedFeeCharge: sdk.OneDec(),
+		},
+		"current sqrt price equals neither and currentSqrtPrice > nextTickSqrtPrice -> fee is the diff between amount specified remaining and amount in": {
+			currentSqrtPrice:         DefaultCurrSqrtPrice,
+			nextTickSqrtPrice:        sqrtBelowDefaultPrice,
+			sqrtPriceLimit:           sqrtBelowDefaultPrice,
+			amountIn:                 sdk.NewDec(101),
+			amountSpecifiedRemaining: sdk.NewDec(102),
+			swapFee:                  onePercentFee,
+
+			// 102 - 101 = 1
+			expectedFeeCharge: sdk.OneDec(),
+		},
+		"current sqrt price equals neither and currentSqrtPrice < nextTickSqrtPrice -> fee is the diff between amount specified remaining and amount in": {
+			currentSqrtPrice:         DefaultCurrSqrtPrice,
+			nextTickSqrtPrice:        sqrtAboveDefaultPrice,
+			sqrtPriceLimit:           sqrtAboveDefaultPrice,
+			amountIn:                 sdk.NewDec(101),
+			amountSpecifiedRemaining: sdk.NewDec(102),
+			swapFee:                  onePercentFee,
+
+			// 102 - 101 = 1
+			expectedFeeCharge: sdk.OneDec(),
```

Comment:
I don't know how I feel about having all test cases expect a zero or one fee. I would feel better having at least one example where the expected fee charge is larger.

Reply:
This isn't a problem..

It's zero when the swap fee is zero. It's one due to how the input parameters are set up. The fee is still calculated to be one given the inputs in each test case.

For each branch in the function, there is a test covering it. I could have chosen a larger value, for example, `103` here to get the result of 2. The test coverage and test completeness would not change from the choice of the value.